state-sync/storage-service/server/src/lib.rs

```diff
@@ -525,40 +525,42 @@ impl StorageReader {
     }
 
     /// Returns the transaction range held in the database (lowest to highest).
-    fn fetch_transaction_range(
+    fn fetch_ledger_range(
         &self,
         latest_version: Version,
     ) -> Result<Option<CompleteDataRange<Version>>, Error> {
-        let first_transaction_version = self
+        let pruning_window_option = self
             .storage
-            .get_first_txn_version()
-            .map_err(|error| Error::StorageErrorEncountered(error.to_string()))?;
-        if let Some(first_transaction_version) = first_transaction_version {
-            let transaction_range =
-                CompleteDataRange::new(first_transaction_version, latest_version)
-                    .map_err(|error| Error::UnexpectedErrorEncountered(error.to_string()))?;
-            Ok(Some(transaction_range))
-        } else {
-            Ok(None)
-        }
-    }
-
-    /// Returns the transaction output range held in the database (lowest to highest).
-    fn fetch_transaction_output_range(
-        &self,
-        latest_version: Version,
-    ) -> Result<Option<CompleteDataRange<Version>>, Error> {
-        let first_output_version = self
-            .storage
-            .get_first_write_set_version()
-            .map_err(|error| Error::StorageErrorEncountered(error.to_string()))?;
-        if let Some(first_output_version) = first_output_version {
-            let output_range = CompleteDataRange::new(first_output_version, latest_version)
-                .map_err(|error| Error::UnexpectedErrorEncountered(error.to_string()))?;
-            Ok(Some(output_range))
+            .get_ledger_prune_window()
+            .map_err(|error| Error::StorageErrorEncountered(error.to_string()))?
+            .map(|window| window as u64);
+        let lowest_transaction_version = if let Some(pruning_window) = pruning_window_option {
+            if latest_version < pruning_window {
+                // No pruning has occured, we can safely assume that the first transaction version is 0
+                0
+            } else {
+                // lowest_account_version = latest_version - pruning_window + 1;
+                latest_version
+                    .checked_sub(pruning_window)
+                    .ok_or_else(|| {
+                        Error::UnexpectedErrorEncountered(
+                            "Lowest account states version has overflown!".into(),
+                        )
+                    })?
+                    .checked_add(1)
+                    .ok_or_else(|| {
+                        Error::UnexpectedErrorEncountered(
+                            "Lowest account states version has overflown!".into(),
+                        )
+                    })?
```

Comment:
I'd just do `latest_version + 1 - pruning_window`..

Reply:
Can you clarify why we don't need to worry about overflow?

Reply:
because your `if` branch has already ruled that out?

Reply:
still `.checked_add(1)` can overflow? Or we are doomed at that point of the time anyways, so we don't need to worry about that :D?

Reply:
exactly..

Reply:
Done